File: snippets/dart_snippets/lib/receive_payment.dart
import 'package:dart_snippets/sdk_instance.dart';
import 'package:breez_sdk/bridge_generated.dart';

Future<ReceivePaymentResponse> receivePayment() async {
  // ANCHOR: receive-payment
  ReceivePaymentRequest req = const ReceivePaymentRequest(
    amountMsat: 3000000,
    description: "Invoice for 3000 sats",
  );
  ReceivePaymentResponse receivePaymentResponse = await breezSDK.receivePayment(req: req);

  print(receivePaymentResponse.lnInvoice);
  // ANCHOR_END: receive-payment

  return receivePaymentResponse;
}

File: snippets/swift/BreezSDKExamples/Sources/ReceivePayment.swift
//
//  ReceivePayment.swift
//  
//
//  Created by ruben on 13/11/2023.
//

import BreezSDK

func receivePayment(sdk: BlockingBreezServices) -> ReceivePaymentResponse? {
    // ANCHOR: receive-payment
    let receivePaymentResponse = try? sdk.receivePayment(
        req: ReceivePaymentRequest(
            amountMsat: 3_000_000,
            description: "Invoice for 3 000 sats"))

    let invoice = receivePaymentResponse?.lnInvoice;
    // ANCHOR_END: receive-payment
    return receivePaymentResponse
}
